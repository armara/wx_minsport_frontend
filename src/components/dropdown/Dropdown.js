import React, { useState, useEffect, useCallback, useMemo } from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'

import DropdownList from 'components/dropdown/dropdownList/DropdownList'
import DropdownTrigger from 'components/dropdown/dropdownTrigger/DropdownTrigger'

import 'components/dropdown/Dropdown.scss'

const Dropdown = ({
  menuId,
  titleKey,
  idKey,
  initialItem,
  items,
  onItemSelect,
  listItemRenderer,
  selectedItemRenderer,
  notFoundPlaceholder,
  listComponent: ListComponent,
  triggerComponent: TriggerComponent,
  triggerComponentProps,
  listComponentProps,
}) => {
  const [isActive, toggleActive] = useState(false)
  const [selectedTitle, setSelectedTitle] = useState(initialItem)

  const changeActiveState = useCallback(() => toggleActive(!isActive), [
    isActive,
    toggleActive,
  ])
  const handleItemSelect = useCallback(
    item => {
      setSelectedTitle(item)
      toggleActive(false)
      onItemSelect(item)
    },
    [setSelectedTitle, toggleActive, onItemSelect]
  )

  const selectedItem = useMemo(() => {
    if (selectedItemRenderer) {
      return selectedItemRenderer(selectedTitle, selectedTitle[titleKey])
    }

    if (selectedTitle[titleKey]) return selectedTitle[titleKey]

    return selectedTitle
  }, [selectedItemRenderer, selectedTitle, titleKey])

  useEffect(() => {
    setSelectedTitle(initialItem)
  }, [initialItem, setSelectedTitle])

  const memoizedOnWindowClick = useCallback(
    event => {
      const { target } = event
      if (target.closest(`#${menuId}`)) return
      if (isActive === true) toggleActive(false)
    },
    [menuId, isActive, toggleActive]
  )
  useEffect(() => {
    if (typeof window === 'undefined') return null

    window.addEventListener('click', memoizedOnWindowClick)
    return () => window.removeEventListener('click', memoizedOnWindowClick)
  }, [memoizedOnWindowClick])

  return (
    <div
      className={classnames(['dropdown', isActive ? 'is-active' : ''])}
      id={menuId}
    >
      <div className="dropdown-trigger">
        <TriggerComponent
          onClick={changeActiveState}
          isActive={isActive}
          selectedItem={selectedItem}
          props={triggerComponentProps}
        />
      </div>
      <div className="dropdown-menu" role="menu">
        <ListComponent
          items={items}
          titleKey={titleKey}
          idKey={idKey}
          onItemSelect={handleItemSelect}
          listItemRenderer={listItemRenderer}
          notFoundPlaceholder={notFoundPlaceholder}
          props={listComponentProps}
        />
      </div>
    </div>
  )
}

Dropdown.defaultProps = {
  initialItem: '',
  titleKey: 'title',
  idKey: 'id',
  onItemSelect: () => {},
  listItemRenderer: undefined,
  selectedItemRenderer: undefined,
  notFoundPlaceholder: undefined,
  listComponent: DropdownList,
  triggerComponent: DropdownTrigger,
  triggerComponentProps: {},
  listComponentProps: {},
}

Dropdown.propTypes = {
  initialItem: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
  listItemRenderer: PropTypes.func,
  selectedItemRenderer: PropTypes.func,
  titleKey: PropTypes.string,
  idKey: PropTypes.string,
  onItemSelect: PropTypes.func,
  notFoundPlaceholder: PropTypes.string,
  listComponent: PropTypes.oneOfType([PropTypes.element, PropTypes.func]),
  triggerComponent: PropTypes.oneOfType([PropTypes.element, PropTypes.func]),
  triggerComponentProps: PropTypes.shape({}),
  listComponentProps: PropTypes.shape({}),
  items: PropTypes.arrayOf(PropTypes.object).isRequired,
  menuId: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
}

export default Dropdown
