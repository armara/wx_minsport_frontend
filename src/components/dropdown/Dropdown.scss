@import "styles/mixins";

.dropdown {
  .dropdown-menu {
    width: 100%;
  }
}