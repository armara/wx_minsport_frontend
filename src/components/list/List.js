import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'

import Image from 'components/image/Image'

import 'components/list/List.scss'

const rowRenderer = item => {
  if (item.type === 'hr') {
    return <hr />
  }

  const { spanClassName } = item

  return (
    <>
      <div>
        {item.image && <Image src={item.image} />}
        <h4>{item.field}</h4>
      </div>

      {!item.value ? (
        '---'
      ) : (
        <span className={spanClassName}>{item.value}</span>
      )}
    </>
  )
}

const List = ({
  id,
  keyId,
  items,
  children,
  className,
  rowClassName,
  getRowClassName,
  notFoundItem,
  childrenProps,
}) => (
  <ul className={classnames([className, 'base-list'])} id={id}>
    {items.length === 0 ? (
      <li className="base-list_not-found">{notFoundItem}</li>
    ) : (
      items.map((item, i) => (
        <li
          key={item[keyId]}
          className={classnames([rowClassName, getRowClassName(item, i)])}
        >
          {children ? children(item, childrenProps) : rowRenderer(item)}
        </li>
      ))
    )}
  </ul>
)

List.defaultProps = {
  id: null,
  keyId: 'key',
  className: '',
  rowClassName: '',
  children: '',
  getRowClassName: () => {},
  childrenProps: {},
  notFoundItem: 'Не найдено',
}

List.propTypes = {
  id: PropTypes.string,
  keyId: PropTypes.string,
  className: PropTypes.string,
  rowClassName: PropTypes.string,
  getRowClassName: PropTypes.func,
  notFoundItem: PropTypes.string,
  items: PropTypes.arrayOf(
    PropTypes.shape({
      key: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
      image: PropTypes.string,
      field: PropTypes.string,
      value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
      type: PropTypes.oneOf(['hr']),
    })
  ).isRequired,
  childrenProps: PropTypes.shape({}),
  children: PropTypes.oneOfType([PropTypes.func, PropTypes.string]),
}

export default List
