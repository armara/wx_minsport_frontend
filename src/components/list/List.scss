@import "styles/mixins";

.base-list {
  .base-list_not-found {
    font-style: italic;
  }

  hr {
    width: 100%;
    margin-top: 15px;
    margin-bottom: 15px;
  }

  li + li {
    margin-top: 15px;
  }

  li {
    display: flex;
    justify-content: space-between;

    h4 {
      @include roboto-regular-text($small, 18px, $evil-gray);
      display: flex;
      align-items: center;
    }

    & > div {
      display: flex;

      img {
        margin-right: 10px;
      }
    }

    & > span {
      @include nav-header-text($small, '18px', $black);
      display: flex;
      align-items: center;
    }
  }
}

.tabs-list {
  padding-top: 0;

  li {
    height: 45px;
  }

  li + li {
    margin-top: 0;
  }
}