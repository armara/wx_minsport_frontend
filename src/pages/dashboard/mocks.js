import React from 'react'
import dayjs from 'utils/day'

import { DAY_FORMAT } from 'utils/localization'
import ContractsTableCell from 'pages/dashboard/components/contractsTableRow/ContractsTableCell'

export const columns = [
  {
    Header: 'Организация',
    id: 'Организация',
    accessor: row => row.organizationName,
  },
  {
    Header: 'Истекает',
    id: 'Истекает',
    accessor: row => dayjs(row.finishDate).format(DAY_FORMAT),
  },
  {
    Header: 'Тип договора',
    id: 'Тип договора',
    accessor: row => (
      <ContractsTableCell
        text={row.contractTitle}
        className={
          !row.contractTitle.includes('безвозмездном')
            ? 'contracts-table-row_paid'
            : 'contracts-table-row_not-paid'
        }
      />
    ),
    style: { whiteSpace: 'unset' },
  },
]

export const data = [
  {
    foundation: 'Федерация баскетбола',
    expire: '15.01.2019',
    status: 'Оплачен',
    options: {
      isPaid: true,
    },
  },
  {
    foundation: 'Федерация баскетбола',
    expire: '12.01.2020',
    status: 'Не оплачен',
    options: {
      isPaid: false,
    },
  },
  {
    foundation: 'Федерация баскетбола',
    expire: '15.04.2020',
    status: 'Не оплачен',
    options: {
      isPaid: false,
    },
  },
  {
    foundation: 'Федерация баскетбола',
    expire: '15.10.2019',
    status: 'Оплачен',
    options: {
      isPaid: true,
    },
  },
  {
    foundation: 'Федерация баскетбола',
    expire: '09.01.2020',
    status: 'Не оплачен',
    options: {
      isPaid: false,
    },
  },
]

export const courtsList = (areas, zones) =>
  areas.map(area => ({
    title: area.name,
    rating: zones
      .filter(el => el.areaId === area.id)
      .reduce((a, b) => a + b.capacity, 0),
    key: area.id,
  }))

export const staffList = coaches =>
  coaches.map(coach => ({
    image: `data:image/jpeg;base64,${coach.avatar}`,
    field: coach.fio,
    key: coach.id,
    id: coach.id,
  }))
