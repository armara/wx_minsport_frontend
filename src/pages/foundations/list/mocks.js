import React from 'react'

import TableCellFoundation from 'components/table/tableCellIcon/TableCellIcon'
import TableCellText from 'components/table/tableCellText/TableCellText'
import TableCellActions from 'components/table/tableCellActions/TableCellActions'

export const columns = [
  {
    id: 'Организация',
    Header: 'Организация',
    isFilterable: true,
    accessor: row => <TableCellFoundation foundation={row.foundation} />,
  },
  {
    id: 'Тип организации',
    Header: 'Тип организации',
    isFilterable: true,
    accessor: row => <TableCellText text={row['foundation-type']} />,
  },
  {
    id: 'Вид организации',
    Header: 'Вид организации',
    isFilterable: true,
    accessor: row => <TableCellText text={row['organization-type']} />,
  },
  {
    id: 'Код организации',
    Header: 'Код организации',
    isFilterable: true,
    accessor: row => <TableCellText text={row['foundation-id']} />,
  },
  {
    id: 'Код / Вид спорта',
    Header: 'Код / Вид спорта',
    isFilterable: true,
    accessor: row => <TableCellText text={row['sport-id']} />,
  },
  {
    id: 'Регистрация',
    Header: 'Регистрация',
    isFilterable: true,
    accessor: row => <TableCellText text={row['registration-date']} />,
  },
  {
    id: 'Действия',
    Header: ' ',
    isFilterable: false,
    accessor: () => <TableCellActions />,
  },
]

export const data = [
  {
    foundation: 'Российская Федерация Баскетбола',
    'foundation-type': 'Федерация',
    'organization-type': 'ШОР',
    'foundation-id': '014 000 Я',
    'sport-id': '152 9013 19 С',
    'registration-date': '12.08.1976',
    actions: '',
  },
  {
    foundation: 'Российская Федерация Баскетбола',
    'foundation-type': 'Федерация',
    'organization-type': 'ШОР',
    'foundation-id': '014 001 Я',
    'sport-id': '152 9013 20 С',
    'registration-date': '12.08.1976',
    actions: '',
  },
  {
    foundation: 'Российская Федерация Баскетбола',
    'foundation-type': 'Федерация',
    'organization-type': 'ШОР',
    'foundation-id': '014 002 Я',
    'sport-id': '152 9013 21 С',
    'registration-date': '12.08.1976',
    actions: '',
  },
  {
    foundation: 'Российская Федерация Баскетбола',
    'foundation-type': 'Федерация',
    'organization-type': 'ШОР',
    'foundation-id': '014 003 Я',
    'sport-id': '152 9013 22 С',
    'registration-date': '12.08.1976',
    actions: '',
  },
]

export const dropdownItems = [
  { title: 'Бюджет', id: 'budget' },
  { title: 'Армия', id: 'forces' },
]
