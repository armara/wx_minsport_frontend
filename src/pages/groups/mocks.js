export const sportsmen = [
  {
    id: '63f268b7-9dcc-49cc-9ee8-8b4bbe84816f',
    title: 'Никита Иванович Панин',
    disciplineTitle: 'Футбол на траве',
    sportTitle: 'Футбол',
    age: '10 лет',
    img: '',
    rankName: 'МСМК',
  },
  {
    id: '63f268b7-9dcc-49cc-2ee8-8b4bbe84816f',
    title: 'Никита Иванович Панин',
    disciplineTitle: 'Футбол на траве',
    sportTitle: 'Футбол',
    age: '10 лет',
    img: '',
    rankName: 'МСМК',
  },
  {
    id: '63f268b7-9dcc-49cc-9ee8-8b46be84816f',
    title: 'Никита Иванович Панин',
    disciplineTitle: 'Футбол на траве',
    sportTitle: 'Футбол',
    age: '10 лет',
    img: '',
    rankName: 'МСМК',
  },
  {
    id: '63f268b7-9dcc-49cc-9ee8-8b43be84816f',
    title: 'Никита Иванович Панин',
    disciplineTitle: 'Футбол на траве',
    sportTitle: 'Футбол',
    age: '10 лет',
    img: '',
    rankName: 'МСМК',
  },
  {
    id: '63f268b7-9dcc-49cc-9ee8-8b4bbg84816f',
    title: 'Никита Иванович Панин',
    disciplineTitle: 'Футбол на траве',
    sportTitle: 'Футбол',
    age: '10 лет',
    img: '',
    rankName: 'МСМК',
  },
  {
    id: '63f268b7-9dcc-49cc-9ee8-8b4bbe84f16f',
    title: 'Никита Иванович Панин',
    disciplineTitle: 'Футбол на траве',
    sportTitle: 'Футбол',
    age: '10 лет',
    img: '',
    rankName: 'МСМК',
  },
  {
    id: '63f268b7-9dcc-49cc-9ee8-8s4bbe84816f',
    title: 'Никита Иванович Панин',
    disciplineTitle: 'Футбол на траве',
    sportTitle: 'Футбол',
    age: '10 лет',
    img: '',
    rankName: 'МСМК',
  },
]

export const groups = [
  {
    agesRange: '18-50 лет',
    disciplineTitle: 'Футбол на траве',
    id: '63f268b7-9dcc-49cc-9ee8-8b4bbe84816f',
    membersCount: 2,
    membersMax: 50,
    price: 100,
    sportTitle: 'Футбол',
    title: 'Футбол (дети из 48 школы)',
  },
  {
    agesRange: '18-50 лет',
    disciplineTitle: 'Футбол на траве',
    id: '63f268b7-39dcc-49cc-9ee8-8b4bbe84816f',
    membersCount: 2,
    membersMax: 50,
    price: 100,
    sportTitle: 'Футбол',
    title: 'Футбол (дети из 4 школы)',
  },
  {
    agesRange: '18-50 лет',
    disciplineTitle: 'Футбол на траве',
    id: '63f268b7-9d2c-49cc-9ee8-8b4bbe84816f',
    membersCount: 2,
    membersMax: 50,
    price: 100,
    sportTitle: 'Футбол',
    title: 'Баскетбол (дети из 17 школы)',
  },
  {
    agesRange: '18-50 лет',
    disciplineTitle: 'Футбол на траве',
    id: '63f268b7-9dcc-49cc-9ee8-8b47be84816f',
    membersCount: 2,
    membersMax: 50,
    price: 100,
    sportTitle: 'Футбол',
    title: 'Плаванье (дети из 34 школы)',
  },
]
