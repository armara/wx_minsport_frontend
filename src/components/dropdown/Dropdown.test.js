import React from 'react'
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'

import Dropdown from 'components/dropdown/Dropdown'

import { dropdownItems } from 'pages/foundations/list/mocks'

configure({ adapter: new Adapter() })

describe('<Dropdown />', () => {
  it('renders correctly default props', () => {
    const tree = renderer
      .create(
        <Dropdown
          items={dropdownItems}
          menuId="Peters Konigsberg Great Embassy"
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('opens on button click', () => {})

  it('renders list correctly', () => {})

  it('closes on outside click', () => {})

  it('closes on button click', () => {})
})
