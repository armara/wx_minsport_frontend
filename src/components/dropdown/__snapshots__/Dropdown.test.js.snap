// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dropdown /> renders correctly default props 1`] = `
<div
  className="dropdown"
  id="Peters Konigsberg Great Embassy"
>
  <div
    className="dropdown-trigger"
  >
    <button
      className="dropdown-trigger button base-button"
      disabled={false}
      id={null}
      onClick={[Function]}
      title=""
      type="button"
    >
      
      <div>
        
      </div>
      <span
        className="icon is-small"
      >
        <i
          aria-hidden="true"
          className="ic-keyboard-arrow-down"
        />
      </span>
    </button>
  </div>
  <div
    className="dropdown-menu"
    role="menu"
  >
    <ul
      className="dropdown-content menu-list"
    >
      <button
        className="dropdown-item button base-button"
        disabled={false}
        id={null}
        onClick={[Function]}
        title=""
        type="button"
      >
        
        Бюджет
      </button>
      <button
        className="dropdown-item button base-button"
        disabled={false}
        id={null}
        onClick={[Function]}
        title=""
        type="button"
      >
        
        Армия
      </button>
    </ul>
  </div>
</div>
`;
